domain/model/service/service.go

```diff
@@ -26,25 +31,71 @@ type State interface {
 // Service defines a service for interacting with the underlying state based
 // information of a model.
 type Service struct {
-	st State
+	st         State
+	toolFinder ToolFinder
+}
+
+// ToolFinder represents a helper for establishing if tools for a specific Juju
+// are available.
+type ToolFinder interface {
+	// HasToolsForVersion will interrogate the tools available in the system and
+	// return nil if tools exist for the specified version. If no tools are
+	// found then a error satisfying [github.com/juju/errors.NotFound] will be
+	// returned. All other errors indicate a problem with parsing the available
+	// tools.
+	HasToolsForVersion(version.Number) error
+}
+
+// toolFinderFn is func type for the ToolFinder interface.
+type toolFinderFn func(version.Number) error
+
+func (t toolFinderFn) HasToolsForVersion(v version.Number) error {
+	return t(v)
 }
 
 // NewService returns a new Service for interacting with a models state.
-func NewService(st State) *Service {
+func NewService(st State, toolFinder ToolFinder) *Service {
 	return &Service{
-		st: st,
+		st:         st,
+		toolFinder: toolFinder,
 	}
 }
 
+// NoopToolFinder is a transition implementation of the tool finder that will
+// return [errors.NotFound] for any version that is not the current controller
+// version.
+// This will be removed and replaced soon.
+func NoopToolFinder() ToolFinder {
```

Comment:
Noop confused me because it does actually do a check. 
Can't think of a significantly better name though (but tools is deprecated).